src/messagingsession/DefaultMessagingSession.ts

```diff
@@ -191,7 +194,7 @@ export default class DefaultMessagingSession implements MessagingSession {
       !this.isClosing &&
       this.canReconnect(event.code) &&
       this.reconnectController.retryWithBackoff(async () => {
-        this.startConnecting(true);
+        await this.startConnecting(true);
```

Comment:
`await`ing this should also not affect as for the successful connection we will be depending on the `messagingSesssionDidStart` right? Tagging Kyu (@simmkyu ) for checking once.

Reply:
this is in the reconnect and want to await so don't have duplicate reconnects going at same time if takes a while to start